src/test/codegen/nounwind-extern.rs

```diff
@@ -2,5 +2,8 @@
 
 #![crate_type = "lib"]
 
+// The `nounwind` attribute does not get added by rustc; it is present here because LLVM
+// analyses determine that this function does not unwind.
```

Comment:
This entire test seems mostly pointless to me. It certainly has nothing to do with which attributes Rust emits, because LLVM optimizations are run.

@alexcrichton you added this test, could you explain?

Reply:
Maybe the test should be changed to make the function panic? Then LLVM couldn't add `nounwind` any more.

Reply:
I think this test is ok. For nightly Rust, `nounwind` here is ok, and should happen even without LLVM optimizations enabled.  For stable Rust, `nounwind` here should only happen if LLVM optimizations deem it safe. 

Adding a test that unconditionally panics looks like a good idea. In nightly Rust, `nounwind` should be applied, and the panic caught and abort. In stable Rust, the function should just not have the `nounwind` attribute.

Reply:
@gnzlbg  The test is ok, sure, but not very useful. And adding or not adding the attribute is already covered by the other tests.

So I think I will just remove this one.